### PR TITLE
Dashboard: show certificate if user has it

### LINKED ISSUE #3862: Dashboard: display "View certificate" link even when user is auditing a course
As a learner who has earned a course certificate, I would like to be able to audit the course again, and still be able to access my certificate. 

Currently, the dashboard message "You are auditing. You need to pay to get credit" is superseding the message "You passed this course. View Certificate" We need some way to display the "View certificate" link in this state. 

See [ZenDesk #15251](https://odl.zendesk.com/agent/tickets/15251) (FA course) and [ZenDesk #15260](https://odl.zendesk.com/agent/tickets/15260) (non-FA course).

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #3862

#### What's this PR do?
Shows the link to certificate for a course if user has it.

Also, fixed a bug where the message about exam was shown even if course didn't have an exam.

#### How should this be manually tested?
Delete folder `output/dashboard_states`
Run ./scripts/test/run_snapshot_dashboard_states.sh --match "taking_again"
Take a look the output

screen shot 2018-03-02 at 2 38 09 pm

### CODE REVIEW

**Comment by noisecapella** [comment]
Thanks for adding the dashboard state for this, it makes manual testing really easy

**Comment by noisecapella** [comment]
Looks like unit tests and linting are still needed, functionality looks good though

**Comment by annagav** [comment]
@noisecapella, fixed all tests!

File: static/js/components/dashboard/courses/StatusMessages.js [hunk 1]
```diff
@@ -169,6 +186,10 @@ export const calculateMessages = (props: CalculateMessagesProps) => {
   ) {
     let message =
       "You are auditing. To get credit, you need to pay for the course."
+    if (course.certificate_url) {
+      message =
```
Comment (noisecapella) [inline static/js/components/dashboard/courses/StatusMessages.js]: Can you add a test for this case?

**Comment by annagav** [comment]
@noisecapella I addressed your comments.

**Pull request merged**